zephyr/lib/cpu.c

```diff
@@ -148,8 +148,7 @@ int cpu_enable_core(int id)
 	 * initialization. By reinitializing the idle thread, we would overwrite the kernel structs
 	 * and the idle thread stack.
 	 */
-	if (!IS_ENABLED(CONFIG_ADSP_IMR_CONTEXT_SAVE) ||
-	    pm_state_next_get(id)->state == PM_STATE_ACTIVE)
+	if (pm_state_next_get(id)->state == PM_STATE_ACTIVE)
```

Comment:
this means, that now if IMR context saving is **not** enabled we potentially call `z_init_cpu()` less frequently, or will it not happen because now the `pm_state_next_get(id)->state == PM_STATE_ACTIVE` will cover all those cases?

Reply:
sure,  we only need to call z_init_cpu() for dsp core init with above condition for both CAVS  and ACE platforms since CAVS dsp core context save is being supported now.  IMR_CONTEXT_SAVE has nothing to do with z_init_cpu because
IMR_CONTEXT_SAVE is for the whole audio context but z_init_cpu is only for dsp core. IMR_CONTEXT_SAVE is not fit for this check but it can differentiate CAVS & ACE, so it was used.